src/main/java/com/kaziamyr/onlinebookstore/mapper/UserMapper.java

```diff
@@ -3,12 +3,29 @@
 import com.kaziamyr.onlinebookstore.config.MapperConfig;
 import com.kaziamyr.onlinebookstore.dto.UserRegistrationRequestDto;
 import com.kaziamyr.onlinebookstore.dto.UserRegistrationResponseDto;
+import com.kaziamyr.onlinebookstore.model.Role;
 import com.kaziamyr.onlinebookstore.model.User;
+import java.util.Set;
+import org.mapstruct.BeforeMapping;
 import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
 
 @Mapper(config = MapperConfig.class)
-public interface UserMapper {
-    UserRegistrationResponseDto toUserResponseDto(User user);
+public abstract class UserMapper {
+    @Autowired
+    private PasswordEncoder passwordEncoder;
 
-    User toModel(UserRegistrationRequestDto requestDto);
+    public abstract UserRegistrationResponseDto toUserResponseDto(User user);
+
+    @BeforeMapping
+    public void addPasswordAndRoles(
+            @MappingTarget User user, UserRegistrationRequestDto requestDto
+    ) {
+        user.setPassword(passwordEncoder.encode(requestDto.getPassword()));
+        user.setRoles(Set.of(new Role(1L, Role.RoleName.ROLE_USER)));
+    }
```

Comment:
Bad mapping. It breaks Single responsibility. Put role inside service, not mapper